src/test/java/com/ncc/aif/TestUtils.java

```diff
@@ -0,0 +1,495 @@
+        // if (valid != expected || forceDump) {
+        if (valid != expected || (forceDump && expected)) {
+            System.out.println("\n----------------------------------------------\n" + testName + "\n\nAIF Model:");
+            RDFDataMgr.write(System.out, model, RDFFormat.TURTLE_PRETTY);
+        }
+
+        // fail if result is unexpected
+        if (valid != expected) {
+            // only print output if there is any
+            if (!valid) {
+                System.out.println("\nFailure:");
+                RDFDataMgr.write(System.out, report.getModel(), RDFFormat.TURTLE_PRETTY);
+            }
+            fail("Validation was expected to " + (expected ? "pass" : "fail") + " but did not");
+        }
+    }
+}
+
+/**
+ * An extension of the TestUtils that supports testing restricted AIF.  Usage and features are the same as TestUtils.
+ */
+class NistTestUtils extends TestUtils {
+
+    /**
+     * Constructor for utilities for testing restricted AIF functionality.
+     *
+     * @param annotationNamespace namespace to use with URIs
+     * @param validator           an AIF validator instantiated based on the caller's ontology and desired NIST restrictions
+     * @param forceDump           whether or not to force dumping of models prior to validation
+     */
+    NistTestUtils(String annotationNamespace, ValidateAIF validator, boolean forceDump) {
+        super(annotationNamespace, validator, forceDump);
+    }
+
+    /**
+     * Call before each test.  Returns a new, empty model with standard AIF namespaces.
+     *
+     * @return a new model with which to start a test; caller may wish to add prefixes for the ontology and annotation
+     */
+    Model startNewTest() {
+        Model model = super.startNewTest();
+        // NIST tests always need type assertion justifications
+        addSourceDocumentToJustification(super.getTypeAssertionJustification(), getDocumentName());
+        return model;
+    }
+
+    Resource makeValidJustification() {
+        return makeValidJustification("sourceDocument");
+    }
+
+    Resource makeValidJustification(String sourceDocument) {
+        Resource justification = super.makeValidJustification();
+        addSourceDocumentToJustification(justification, sourceDocument);
+        return justification;
+    }
+
+    /**
+     * Makes and returns a valid NIST-restricted entity of the specified type and its cluster.
+     *
+     * @param type entity type
+     * @return a key-value Pair of the entity Resource (key) and its associated cluster Resource (value)
+     */
+    Pair<Resource, Resource> makeValidNistEntity(Resource type) {
+        Resource entity = makeEntity(model, getEntityUri(), system);
+        markJustification(addType(entity, type), getTypeAssertionJustification());
+        Resource entityCluster = makeClusterWithPrototype(model, getClusterUri(), entity, system);
+        return new Pair<>(entity, entityCluster);
```

Comment:
I would recommend putting this and the two lines above it into a common private static method to reduce duplicate code

Reply:
It's funny you should mention that.  Somewhere along the way I noticed the same thing as you (the argument also applies to `TestUtiils` and `NistHypothesisTestUtils` although there's less code redundancy there) and I thought maybe I should just create one generic API call, `makeValidObject(Enum)` or something where you'd pass an Enum value of, essentially, ENTITY, EVENT, or RELATION.  Then it would call the appropriate private method, or perhaps if short enough, just switch off the Enum and make the call right there.  WDYT-- bad idea, too late, or go for it?

But if I don't make the above change, I can certainly do as you suggest and pull those three lines into a private method in `NistTestUtils`.

Reply:
I don't think it's a bad idea or too late. As you've already done the hard work of separating the APIs, I'll leave it up to you whether you'd rather consolidate everything into a single method or just make a private, re-usable method.

Reply:
Well, the hard work is updating all of the `XXXExamplesAndValidationTest` Java files to use the new API and call `makeValidObject(TestUtils.ENTITY)` and `makeValidNistObject(TestUtils.ENTITY)`.  In theory, then, we could even make the current `makeValidEntity()` (etc.) API private...

And actually, this approach doesn't work for NistHypothesisTestUtils because entities take a String handle and events and relations take a double importance (unless we cheat and have them pass their doubles as Strings).

Reply:
I have moved those three lines to a private helper method, `makeValidNistObject()`.